utils/s2n_ensure.c

```diff
@@ -15,11 +15,9 @@
 
 #include "utils/s2n_safety.h"
 
-void *s2n_ensure_memcpy_trace(void *restrict to, const void *restrict from, size_t size, const char *debug_str)
+void *s2n_ensure_memcpy_trace(void *restrict to, const void *restrict from, size_t size)
 {
     if (to == NULL || from == NULL) {
-        s2n_errno = S2N_ERR_NULL;
-        s2n_debug_str = debug_str;
         return NULL;
     }
```

Comment:
I know you swapped the macro from doing GUARD to ENSURE_REF so that we don't need to set s2n_errno here, but I think we probably still should just in case. Worst case, the error location is marked as s2n_ensure.c rather than where __S2N_ENSURE_SAFE_MEMCPY was called, but at least there's a valid s2n_errno. We can just use PTR_ENSURE_REF here rather than setting s2n_errno manually. The same could be said for memmove-- we probably need to PTR_ENSURE its result.

Or at least we should put a comment on this method warning not to PTR_GUARD it, but that hasn't worked well in the past ;)